library/alloc/src/vec/drain.rs

```diff
@@ -128,10 +128,6 @@ impl<T, A: Allocator> Drop for Drain<'_, T, A> {
 
         let iter = mem::replace(&mut self.iter, (&mut []).iter());
         let drop_len = iter.len();
-        let drop_ptr = iter.as_slice().as_ptr();
-
-        // forget iter so there's no aliasing reference
-        drop(iter);
```

Comment:
Does `iter` have an actual `impl Drop` that used to be run here?

Reply:
No, it wasn't actually necessary. I thought `IntoIter` internally holds a slice reference that would be invalidated by mutating the vec through other pointers. But it actually contains pointers.

Reply:
Also, at least under Stacked Borrows, `drop`ing a reference doesn't help with anything -- it only marks the reference as *active* the moment it gets dropped.